### PR TITLE
Run tests on Dockerfiles.

### PR DESCRIPTION
**Pull request by schmmd**

### CODE REVIEW

**Comment by matt-gardner** [comment]
Will this make it so that the tests run every time you use `run_on_aws.sh` (or equivalent, whenever you've changed that)?

**Comment by matt-gardner** [comment]
This runs the tests while _building_ the docker image, not when running it?  That's better than it could be...  I'd still be happier with this if there were a flag I could pass to not run the tests, and we used that flag in the `run_on_aws` script.

**Comment by matt-gardner** [comment]
The amount of time the tests take to run is only going to grow, and it could end up being a not-insignificant chunk of time.

**Comment by schmmd** [comment]
Yes, this runs the tests while building the docker image not while running the docker image.  If this becomes onerous over time, we could:

* Remove running the tests.
* Put this behind a flag.
* Speed up our tests.

**Comment by schmmd** [comment]
Actually, when tests start taking too long, we should just run pytest on a subset in our Docker build.  Presently however, the tests do not take a significant amount of time.

File: Dockerfile.cpu [hunk 1]
```diff
@@ -36,6 +36,7 @@ ENV PYTHONHASHSEED 2157
 ENV LC_ALL=C.UTF-8
 ENV LANG=C.UTF-8
 
+# Copy select files needed for installing requirements
```
Comment (joelgrus) [inline Dockerfile.cpu]: please add an explanation of why we're doing this

**Pull request merged**